### ISSUE TITLE
vnc_auto.html crash

### ISSUE THREAD

**Issue by samhed** [kanaka/noVNC#646]
vnc_auto.html doens't work after commit 9104d3b since it seems to depend on RFB._init_msg() to not return anything. I think it's bad practice to have functions be inconsistent with return values. Any ideas on the thought behind this @kanaka or @DirectXMan12 ?

**Comment by kanaka** [comment]
Did you mean 4e0c36dda7?

I think the issue is that in the 'password' state case, we are basically waiting for user input and so _init_msg ignores that state. I think that's probably the only additional case that we need to handle like that, but this part of the code is getting pretty fuzzy for me so I can't say for sure.

This change makes it work for me:

```patch
g diff include/rfb.js
diff --git a/include/rfb.js b/include/rfb.js
index 9935962..ce5b68a 100644
--- a/include/rfb.js
+++ b/include/rfb.js
@@ -1036,6 +1036,9 @@ var RFB;
                 case 'ServerInitialisation':
                     return this._negotiate_server_init();
 
+                case 'password':
+                    return true;
+
                 default:
                     return this._fail("Unknown state: " + this._rfb_state);
             }
```

**Comment by samhed** [comment]
Yeah, that's the one I meant! Okay, that sounds ok to me

**Comment by samhed** [comment]
Actually, after looking further into this, why do we even have a 'password'-state? We have the 'onPasswordRequired' callback

**Comment by kanaka** [comment]
@samhed see https://github.com/kanaka/noVNC/pull/648#issuecomment-244105808 for separate 'password' state rationale.

**Comment by samhed** [comment]
Fixed by 7d714b1
